plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue

```diff
@@ -0,0 +1,292 @@
+        // We need to find a proper solution for this. A workaround for now could be an event or
+        // something to let other components render a specific page.
+        this.loading = true;
+        const element = $('[piwik-dashboard]');
+        const scope = angular.element(element).scope() as any; // eslint-disable-line
+        scope.fetchDashboard(parseInt(subcategory, 10)).then(() => {
+          this.loading = false;
+        }, () => {
+          this.loading = false;
+        });
+
+        return;
+      }
+
+      ReportingPageStoreInstance.fetchPage(category, subcategory).then(() => {
+        const hasNoPage = !ReportingPageStoreInstance.page.value;
+        if (hasNoPage) {
+          const page = ReportingPagesStoreInstance.findPageInCategory(category);
+          if (page && page.subcategory) {
+            MatomoUrl.updateHash({
+              ...MatomoUrl.hashParsed.value,
+              subcategory: page.subcategory.id,
+            });
+            return;
+          }
+        }
+
+        this.hasNoPage = hasNoPage;
+        this.loading = false;
+      });
+    },
+    renderInitialPage() {
+      const parsed = MatomoUrl.parsed.value;
+      this.renderPage(parsed.category, parsed.subcategory);
+    },
+    showOnlyRawDataMessageIfRequired() {
+      if (this.hasRawData && this.hasNoVisits) {
+        showOnlyRawDataNotification();
+      }
+
+      const parsedUrl = MatomoUrl.parsed.value;
+
+      const { segment } = parsedUrl;
+      if (segment) {
+        hideOnlyRawDataNoticifation();
+        return;
+      }
+
+      const subcategoryExceptions = [
+        'Live_VisitorLog',
+        'General_RealTime',
+        'UserCountryMap_RealTimeMap',
+        'MediaAnalytics_TypeAudienceLog',
+        'MediaAnalytics_TypeRealTime',
+        'FormAnalytics_TypeRealTime',
+        'Goals_AddNewGoal',
+      ];
+
+      const categoryExceptions = [
+        'HeatmapSessionRecording_Heatmaps',
+        'HeatmapSessionRecording_SessionRecordings',
+        'Marketplace_Marketplace',
+      ];
+
+      if (subcategoryExceptions.indexOf(parsedUrl.subcategory) !== -1
+        || categoryExceptions.indexOf(parsedUrl.category) !== -1
+        || parsedUrl.subcategory.toLowerCase().indexOf('manage') !== -1
+      ) {
+        hideOnlyRawDataNoticifation();
+        return;
+      }
+
+      const minuteInMilliseconds = 60000;
+      if (this.dateLastChecked
+        && (new Date().getTime() - this.dateLastChecked) < minuteInMilliseconds
+      ) {
+        return;
+      }
+
+      AjaxHelper.fetch({ method: 'VisitsSummary.getVisits' }).then((json) => {
+        this.dateLastChecked = new Date().getTime();
+
+        if (json.value > 0) {
+          this.hasNoVisits = false;
+          hideOnlyRawDataNoticifation();
+          return;
+        }
+
+        this.hasNoVisits = true;
+
+        if (this.hasRawData) {
+          showOnlyRawDataNotification();
+          return;
+        }
+
+        AjaxHelper.fetch({
```

Comment:
(just suggestion doesn't affect the performance or function) it seems like ajax is waiting another ajax finished. maybe we can use await.
```javascript
   async showOnlyRawDataMessageIfRequired() {
      try {
        const json = await AjaxHelper.fetch({ method: 'VisitsSummary.getVisits' });
      } catch (e) {
      }
    },
```

Reply:
I'll look into using .then(). FYI, the Vue migration doesn't support using async/await since the regenerator runtime is not added (it would have to be added to every UMD not just MatomoPolyfills). There are details in one of the first [Vue] prs.